specification/ai/Azure.AI.Projects/agents/tools/models.tsp

```diff
@@ -437,3 +460,33 @@ model ToolConnection {
   @encodedName("application/json", "connection_id")
   connectionId: string;
 }
+
+// Azure function bindings
+@discriminator("type")
+@doc("The Azure function binding")
+model AzureFunctionBinding {
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+  @doc("The type of binding.")
+  type: string;
+}
+
+@doc("The definition of input or output queue for Azure function.")
+model AzureStorageQueueBinding extends AzureFunctionBinding {
+  @doc("The type of binding, which is always 'storage_queue'.")
+  type: "storage_queue";
+
+  @doc("Storage queue.")
+  @encodedName("application/json", "storage_queue")
+  StorageQueue: AzureFunctionStorageQueue;
+}
+
+@doc("The structure for keeping storage queue name and URI.")
+model AzureFunctionStorageQueue {
+  @doc("URI to the Azure Storage Queue service allowing you to manipulate a queue.")
+  @encodedName("application/json", "queue_service_uri")
+  StorageQueueUri: string;
```

Comment:
```suggestion
  storageQueueUri: string;
```

Reply:
Why are we choosing different names between the wire and the service definition (encoded names)? It is fine to swap between snake_case and camelCase, but for anything else, being different is a clear negative as it will bleed out in docs, service generated errors etc.

Reply:
I think, the primary reason is that the variables in camelCase are preferrable in C#/JS, from the other side, code generator for Python (where snake_case is more preferrable), will still convert the names. Probably, code generator for these languages will not convert them.

Reply:
Fixed

Reply:
This delta goes beyond snake <-> camel casing though - `queue_service_uri` to `storageQueueUri`.

Also, we should not use the term `uri`. For storage accounts specifically, they are referred to as endpoints.

Reply:
The REST API should use endpoint suffix. I don't think queue endpoints are refererred to as anything other than endpoints elsewhere in Azure.

Reply:
Fixed.